server/src/main/java/org/elasticsearch/action/admin/indices/segments/IndicesSegmentResponse.java

```diff
@@ -15,23 +15,27 @@
 import org.apache.lucene.util.Accountable;
 import org.elasticsearch.action.support.DefaultShardOperationFailedException;
 import org.elasticsearch.action.support.broadcast.BroadcastResponse;
+import org.elasticsearch.common.collect.Iterators;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.xcontent.ChunkedToXContent;
 import org.elasticsearch.core.RestApiVersion;
 import org.elasticsearch.index.engine.Segment;
-import org.elasticsearch.transport.Transports;
+import org.elasticsearch.rest.action.RestActions;
+import org.elasticsearch.xcontent.ToXContent;
 import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 
-public class IndicesSegmentResponse extends BroadcastResponse {
+public class IndicesSegmentResponse extends BroadcastResponse implements ChunkedToXContent {
```

Comment:
Hmm this is another case of something that implements both chunked and unchunked interfaces, risking accidental de-chunking (and see also that duplicated call to `buildBroadcastShardsHeader`).

Could we instead make `BroadcastResponse` always use chunked responses and change `addCustomXContentFields` to something chunking-compatible? That'd fix this case plus indices stats, searchable snapshot stats, data stream stats, field usage stats, disk usage stats, ...